src/libraries/System.ComponentModel.TypeConverter/tests/Design/DesigntimeLicenseContextSerializerTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.IO;
+using System.Reflection;
+using Microsoft.DotNet.RemoteExecutor;
+using Xunit;
+
+namespace System.ComponentModel.Design.Tests
+{
+    public static class DesigntimeLicenseContextSerializerTests
+    {
+        private static string enableBinaryFormatterInTypeConverter = "System.ComponentModel.TypeConverter.AllowBinaryFormatterUsageInTypeConverterSerialization";
+        private static string enableBinaryFormatter = "System.Runtime.Serialization.EnableUnsafeBinaryFormatterSerialization";
+
+        public static bool AreBinaryFormatterAndRemoteExecutorSupportedOnThisPlatform => PlatformDetection.IsBinaryFormatterSupported && RemoteExecutor.IsSupported;
+
+        [ConditionalFact(nameof(AreBinaryFormatterAndRemoteExecutorSupportedOnThisPlatform))]
+        public static void SerializeAndDeserialize()
+        {
+            RemoteExecutor.Invoke(() =>
+            {
+                foreach (var (useBinaryFormatter, key) in new System.Collections.Generic.List<Tuple<bool, string>>() {
```

Comment:
If you stop using LocalAppContextSwitches, you will need to pull this for loop outside of the `RemoteExecutor.Invoke`.

Reply:
Why? If I pull it out, RemoteExecutor complains with `Field marshaling is not supported by Invoke`.

Reply:
Because with the "simple" approach it will only read the AppContext setting one time and cache it for the rest of process.